src/Hedgehog/Extras/Stock/String.hs

```diff
@@ -21,3 +28,7 @@ lastLine = strip . L.unlines . L.reverse . L.take 1 . L.reverse . L.lines
 -- | Get the first line in the string
 firstLine :: String -> String
 firstLine = strip . L.unlines . L.take 1 . L.lines
+
+-- | Read the string into a value
+readM :: (Read a, Show a, H.MonadTest m, MonadCatch m, HasCallStack) => String -> m a
+readM = withFrozenCallStack . H.noteShowM . H.evalEither . readEither . strip
```

Comment:
What's the call to `strip` for?

Reply:
I had newlines in the output, so I needed to trim the string before reading it. I can add a comment warning about that.